components/oauth-extensions/dynamic-client-manager/src/main/java/org/wso2/carbon/identity/oauth/extension/impl/ClientRegistrationServiceImpl.java

```diff
@@ -75,6 +77,22 @@ public Response register(RegistrationProfile profile) {
         }
     }
 
+    @DELETE
+    @Override
+    public Response unregister(UnregistrationProfile profile) {
+        String applicationName = profile.getApplicationName();
+        String consumerKey = profile.getConsumerKey();
+        String userId = profile.getUserId();
+        try {
+            this.unregisterApplication(userId, applicationName, consumerKey);
```

Comment:
This is fixed in commit [1].
[1] https://github.com/wso2/carbon-device-mgt/commit/8073d46f58c8eede4d1960b026cc32b7b23b28cb